src/components/Dialogs/Conditions/OptionSetCondition.js

```diff
@@ -78,13 +78,18 @@ const OptionSetCondition = ({
     const dataTest = 'option-set'
 
     const setValues = (selected) => {
-        addMetadata(
-            state.options
-                .filter((item) => selected.includes(item.code))
-                .map((item) => ({
+        state.options
+            .filter(
+                (item) =>
+                    selected.includes(item.code) &&
+                    !selectedOptions.find((so) => so.code === item.code)
```

Comment:
As `Transfer` returns all selected options, even when only a subset has been added (e.g. if `1` is already selected and `2` is selected by clicking, then both `1` and `2` will be returned as selected), this prevents `addMetadata` to be fired for "old" selected options (`1` in the example above)

Before: `1` is preselected, `2` is now selected -> `addMetadata(1), addMetadata(2)`
Now: `1` is preselected, `2` is now selected -> `addMetadata(2)`